### ISSUE TITLE
Killedg in train batch

### ISSUE THREAD

**Issue by berryweinst** [facebookresearch/dlrm#4]
Hi, 
After processing the train files, I get this "killedge" and the counting stops. Can you please advice?

Running:
python dlrm_s_pytorch.py --arch-sparse-feature-size=16 --arch-mlp-bot="13-512-256-64-16" --arch-mlp-top="512-256-1" --data-generation=dataset --data-set=kaggle --processed-data-file=./kaggle_data/kaggleAdDisplayChallenge_processed.npz --loss-function=bce --round-targets=True --learning-rate=0.1 --mini-batch-size=128 --print-freq=1024 --print-time --raw-data-file ~/criteo/train.txt --use-gpu

Using 4 GPU(s)...
Loading kaggle dataset...
Reading dataset from ./kaggle_data/kaggleAdDisplayChallenge_processed.npz
Defined training and testing indices...
Randomized indices...
Split data according to indices...
Converted to tensors...done!
Sparse features = 26, Dense features = 13
Training data
Total number of batches 263115
Killedg in train batch: 127687 / 263115

**Comment by dmudiger** [comment]
Looks the application got killed during the data-perp phase, can you please check if there were any system errors, that could have caused this ?

**Comment by mnaumovfb** [comment]
Please try running with a single GPU by setting environment variable CUDA_VISIBLE_DEVICES. 

The environment variable CUDA_VISIBLE_DEVICES controls which GPU(s) will be used for experiments. For instance, if the desired GPU id for experiment is 0, in bash shell you can write:
> export CUDA_VISIBLE_DEVICES=0

The code does provide a functionally correct custom parallel implementation on multiple GPUs, but it has many overheads and it probably will not work on a realistic dataset at the moment. However, it can be run in standalone mode as shown in the benchmarking script
> bench/dlrm_s_benchmark.sh

**Comment by berryweinst** [comment]
Thanks. One device conf didn't make it to crash.

**Comment by dmudiger** [comment]
The latest commit should also work with multiple-devices (with the right environmental settings, "export CUDA_VISIBLE_DEVICES=0,1,2,3") 
Looks like the error you had before was potentially from a memory-leak in the data pre-processing. 
Can you please try the latest commit with multiple devices ?

**Comment by berryweinst** [comment]
Sure, I'll try that.
Thanks.

**Comment by mnaumovfb** [comment]
I'm glad it worked. Please keep in mind that the parallel code is functionally correct, but it is not optimized for performance. I would still recommend using a single GPU for your experiments, unless you are working directly on improving the parallel performance of the code.
